xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -81,6 +82,28 @@ const (
 )
 
 func validateClusterAndConstructClusterUpdate(cluster *v3clusterpb.Cluster) (ClusterUpdate, error) {
+	stringMD := make(map[string]string)
+	if fmd := cluster.GetMetadata().GetFilterMetadata(); fmd != nil {
+		if val, ok := fmd["com.google.csm.telemetry_labels"]; ok {
+			/*
+				"service_name" = "",
+				"service_namespace" = ""
+			*/
+			if fields := val.GetFields(); fields != nil {
+				if val, ok := fields["service_name"]; ok {
+					if _, isStringVal := val.GetKind().(*structpb.Value_StringValue); isStringVal {
```

Comment:
s/isStringVal/ok/ .. `ok` is the de-facto standard for type assertion checks unless you're doing something especially complex.  Yes, it's nested, but that's fine since you never need the outer `ok` again.